spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -56,15 +68,61 @@ public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
 		this.executorProvider = executorProvider;
 		this.channelProvider = channelProvider;
 		this.credentialsProvider = credentialsProvider;
+		this.pullTimeoutMillis = pullTimeoutMillis;
+	}
+
+	/**
+	 * Enables simple a timeout for subscription pull requests.
+	 *
+	 * @param pullTimeoutMillis Timeout in milliseconds for subscription pull requests. If
+	 * set, disables pull retry logic and uses the simple timeout. If set to null, default
+	 * pull behavior with retries will be used.
+	 */
+	public void setPullTimeoutMillis(Long pullTimeoutMillis) {
+		this.pullTimeoutMillis = pullTimeoutMillis;
 	}
 
 	@Override
-	public Subscriber getSubscriber(String subscriptionName, MessageReceiver receiver) {
+	public Subscriber createSubscriber(String subscriptionName, MessageReceiver receiver) {
 		return Subscriber.newBuilder(
 				SubscriptionName.of(this.projectId, subscriptionName), receiver)
 				.setChannelProvider(this.channelProvider)
 				.setExecutorProvider(this.executorProvider)
 				.setCredentialsProvider(this.credentialsProvider)
 				.build();
 	}
+
+	@Override
+	public PullRequest createPullRequest(String subscriptionName, Integer maxMessages,
```

Comment:
It's weird for a `SubscriberFactory` to be generating a `PullRequest`, but I guess it's the best option we have because they take in a project ID?
How about creating a `PullRequestFactory`? Too overkill?

Reply:
I think it's OK to just have on factory for publishing and one for consuming/subscribing.